contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,85 @@
+# VueJs
+
+Author: Lu Lechuan
+
+A JavaScript framework is an application framework written in JavaScript. It differs from a JavaScript library in its control flow: a library offers functions to be called by its parent code, whereas a framework defines the entire application design.
+JavaScript frameworks aim to improve code quality and maintainablity, example of some javascript frameworks include Angular, React and Vue.
+
+## Advantages of VueJs
+VueJs is an approachable, versetile and performant framework that helps to create a more maintainable and testable database.
+
+* approachable: VueJs is very easy to learn. Comparing to other framework such as Augular, Vue is simple in term of API and design. Learning enough to build non-trivial applications typically takes less than a day.
+
+* progressive: Vue is designed from the ground up to be incrementally adoptable. The core library is focused on the view layer only, and is easy to pick up and integrate with other libraries or existing projects. This means that if you hava a large application, you can plug Vue into just a part of your application without messing the rest up.
+
+* versetile: Vue is perfectly capable of powering sophisticated Single-Page Applications when used in combination with modern tooling and supporting libraries.
+
+* clean: Vue symtax is simple and this can make the HTML pages very clean.
+
+## Installation
+
+There are many ways to install VueJs.
+* Direct `<script>` Include:
+  - [development version](https://vuejs.org/js/vue.js)
+  - [product version](https://vuejs.org/js/vue.min.js)
+* CDN:
+  - [cdn](<script src="https://cdn.jsdelivr.net/npm/vue@2.5.13/dist/vue.js"></script>)
+* NPM:
+  - $ npm install vue
+* CLI:
+```cli
+    # install vue-cli
+    $ npm install --global vue-cli
+    # create a new project using the "webpack" template
+    $ vue init webpack my-project
+    # install dependencies and go!
+    $ cd my-project
+    $ npm install
+    $ npm run dev
+```
+
+Apart from installing VueJs, you can install VueJs development tools in your browser. This will allow you to inspect and debug the Vue components in your projects. The link is as follow:
+- [Vue DevTools](https://github.com/vuejs/vue-devtools#vue-devtools)
+
+## The first usage of VueJs:
```

Comment:
My intention is to add a Vue version of "Hello World" so that others can know how to start. I wonder if this is a project? 
And yes thanks for the advice, I will add links to some Vue tutorials.